receivers/discord/discord.go

```diff
@@ -219,66 +217,49 @@ func (d Notifier) SendResolved() bool {
 
 func (d Notifier) constructAttachments(ctx context.Context, alerts []*types.Alert, embedQuota int) []discordAttachment {
 	attachments := make([]discordAttachment, 0, embedQuota)
-	embedsUsed := 0
-	for _, alert := range alerts {
-		// Check if the image limit has been reached at the start of each iteration.
-		if embedsUsed >= embedQuota {
-			d.log.Warn("Discord embed quota reached, not creating more attachments for this notification", "embedQuota", embedQuota)
-			break
-		}
+	_ = images.WithStoredImages(ctx, d.log, d.images,
+		func(index int, image images.Image) error {
+			// Check if the image limit has been reached at the start of each iteration.
+			if len(attachments) >= embedQuota {
+				d.log.Warn("Discord embed quota reached, not creating more attachments for this notification", "embedQuota", embedQuota)
+				return images.ErrImagesDone
+			}
 
-		attachment, err := d.getAttachment(ctx, alert)
-		if err != nil {
-			if errors.Is(err, images.ErrNoImageForAlert) {
-				// There's no image for this alert, continue.
-				continue
+			alert := alerts[index]
+			attachment, err := d.getAttachment(ctx, alert, image)
+			if err != nil {
+				d.log.Error("failed to create an attachment for Discord", "alert", alert, "error", err)
+				return nil
 			}
-			d.log.Error("failed to create an attachment for Discord", "alert", alert, "error", err)
-			continue
-		}
 
-		// We got an attachment, either using the image URL or bytes.
-		attachments = append(attachments, attachment)
-		embedsUsed++
-	}
+			// We got an attachment, either using the image URL or bytes.
+			attachments = append(attachments, *attachment)
+			return nil
+		}, alerts...)
 
 	return attachments
```

Comment:
```suggestion
        if err != nil {
		d.log.Error("failed to create attachments for Discord", "error", err)
	}
	return attachments
```

Only return the attachments if you're sure it's safe to do so, otherwise return `nil` on error.